Version v11.1.2

danjm opened the pull request.

zone-live committed:
[MMI] - Updates the mmi packages, containing a small fix (#21164)

Updates the MMI packages, specifically the SDK one that contains a fix
regarding the content-type header of a client api being set wrongly.

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

- [x] Manual testing (e.g. pull and build branch, run in browser, test
code being changed)
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR
to the release milestone

If further QA is required (e.g. new feature, complex testing steps,
large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

danjm committed:
v11.1.2

zone-live commented:
@metamaskbot update-policies

zone-live commented:
@metamaskbot update-policies

danjm commented on ui/pages/institutional/custody/custody.js:
```diff
@@ -597,8 +597,11 @@ const CustodyPage = () => {
                 history.push({
                   pathname: CUSTODY_ACCOUNT_DONE_ROUTE,
                   state: {
-                    imgSrc: selectedCustodian.iconUrl,
```
@zone-live These changes were not in the PR that was cherry-picked into this branch. Are they supposed to be here? And if so is there any reason they are included here and not on develop?

zone-live replied:
They need to be here as well, in develop they are [here](https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/institutional/custody/custody.js#L635) 👍🏼 
@danjm

danjm replied:
Instead of adding these changes on this branch, can we cherry-pick the commit that added them to develop?

zone-live replied:
I thought about it, but it would introduce more changes that we don't want here for now doing that. It's just this bit that is needed.

danjm commented on package.json:
```diff
@@ -223,14 +223,14 @@
     "@lavamoat/snow": "^1.5.0",
     "@material-ui/core": "^4.11.0",
-    "@metamask-institutional/custody-controller": "^0.2.10",
-    "@metamask-institutional/custody-keyring": "^0.0.27",
```
@zone-live The cherry-picked PR updated the custody keyring to v1.0.3, but this PR does not. Is this intended?

zone-live replied:
Yeah, the cherry-picked PR didn't think we would go back to 11.1 😆 so I had to decrease those two packages versions to be the same ones we have in 11.1.

zone-live replied:
Meaning, the original PR just had the most up to date versions because it went to be merged against `develop` and since develop has the latest changes, that works.
But since this PR is going against `master` and doesn't have the latest changes we have in develop, those two packages versions have to stay the same.

Don't know if this answered the question 😅

danjm replied:
That makes sense. I just want to make sure that the issues you need to fix don't rely on v1.0.3 of `custody-keyring`

danjm committed:
Add lavamoat policy and yarn version changes need for v11.1.2

danjm committed:
Add UI change needed for v11.1.2 which was part of 452e177 (PR #20982)

danjm committed:
Revert test additions from 16a0d16ea (#21177)

## **Description**
As part of PR https://github.com/MetaMask/metamask-extension/pull/21033,
I made commit 16a0d16ea

This introduced a test assertion that fails periodically on firefox. It
consistently passes on chrome in CI, and passes when I run locally in
firefox and chrome. I have not been able to properly debug why these
additions were flaky.

In the interests of removing friction from our build and release flows,
I am going to revert those changes for now, and then we can further
investigate later (issue for that is here
https://github.com/MetaMask/metamask-extension/issues/21178)

## **Manual testing steps**

e2e tests should pass, no manual tests needed.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

brad-decker committed:
fix(#21175): safe access of image (#21176)

## **Description**
Fixes #21175 . Found a regression during personal usage of the
extension. @jiexi already fixed this in a PR #20916 to update the
assets-controller but this single line change is being pulled forward
for a hotfix

## **Manual testing steps**
Use steps from issue which are:
1. Install the extension (develop and then this branch)
2. Import a wallet with NFTs
3. Enable autodetection and open sea
4. See the homescreen won't load and error is cannot access startsWith
of undefined. On this branch that is resolved

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

danjm committed:
Ignored ripple-lib deprecation advisory, as we do not depend on it in MetaMask (#21182)

## **Description**
This PR addresses a `yarn audit` failure that we currently see on
develop. The failure is due to the deprecation of the `ripple-lib`
library used within `@trezor/blockchain-link@npm:2.1.8`. Although
imported in one of our dependencies, MetaMask does not depend on that
library, and it's code is not used by our codebase. We can safely ignore
this advisory.

## **Manual testing steps**

Not needed. Builds should pass without a `yarn audit` warning

danjm commented:
I have QA'd the NFT behaviour on this branch, and this fixes the issue we are seeing in prod.
